### PR TITLE
feat: Avatar's aria label includes 'active' or 'inactive' when using the active prop

### LINKED ISSUE #23128: Avatar: example naming, badge, and active ring
Avatar has a few accessibility bugs showing up in storybook:

1. The first avatar in the default story needs to have an accName exposed
2. The avatars in the "Active" story should be given realistic names so that the active/inactive state is exposed separately from the name
3. The avatars in the "Active Appearance" story need to have their active state exposed (this would be a component change rather than a story change)
4. The badges in the badge story have an `aria-label` applied to a generic `<span>` element -- that prop should be removed from the story code
5. The badges need a role of `img` and to not be hidden so the accName is correctly calculated and not relying on browser author correction mechanisms.

### PR DESCRIPTION
**Pull request by behowell**
## Current Behavior

The state of Avatar's `active` prop is conveyed visually via a highlight ring, but not conveyed to accessibility tools.

## New Behavior

Append "active" or "inactive" to the aria label when the `active` prop is set. 
* When the Avatar is using the `aria-label` prop, simply append the string to it.
* If the Avatar needs to use `aria-labelledby`, then render a hidden span with the active label, and append the span's ID to `aria-labelledby`

There is no prop to control the strings yet; they are hardcoded for now. In the future, they will be able to be localized once we implement an i18n solution.

## Related Issue(s)

* Fixes #23128

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/483f160617f1037e672fe74b8e52abfee9378517'>483f160617f1037e672fe74b8e52abfee9378517</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24901/perf-test-react-components/Avatar-mount_base.html">1213</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24901/perf-test-react-components/Avatar-mount.html">1300</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24901/perf-test-react-components/Button-mount_base.html">841</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24901/perf-test-react-components/Button-mount.html">844</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24901/perf-test-react-components/FluentProvider-mount_base.html">1466</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24901/perf-test-react-components/FluentProvider-mount.html">1449</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24901/perf-test-react-components/FluentProviderWithTheme-mount_base.html">614</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24901/perf-test-react-components/FluentProviderWithTheme-mount.html">581</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24901/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">521</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24901/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">587</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24901/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">560</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24901/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">624</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24901/perf-test-react-components/MakeStyles-mount_base.html">1713</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24901/perf-test-react-components/MakeStyles-mount.html">1733</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24901/perf-test-react-components/SpinButton-mount_base.html">2384</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/24901/perf-test-react-components/SpinButton-mount.html">2385</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: packages/react-components/react-avatar/src/components/Avatar/Avatar.types.ts [hunk 1]
```diff
@@ -32,6 +32,17 @@ export type AvatarSlots = {
...
+  /**
+   * Hidden text that is appended to the generated aria-labelledby to describe the active state.
+   * This will be ignored if a custom `aria-label` or `aria-labelledby` is set on this Avatar.
+   *
+   * The default value depends on the `active` prop:
+   * * unset: `undefined`
+   * * active: `strings.active`
+   * * inactive: `strings.inactive`
+   */
+  activeAriaLabel?: Slot<'span'>;
```
[the remark this answers is not in the document]
Comment (behowell) [reply packages/react-components/react-avatar/src/components/Avatar/Avatar.types.ts]: That's a good point. I've updated the PR to not use a slot, and just hardcode a hidden span for the active label if it's needed. I also updated it to append to the `aria-label` if possible (if it's not using `aria-labelledby`).

**Review by smhigley — APPROVED** [review]
I like it!

**Review by sopranopillow — APPROVED** [review]

**Pull request merged**